addons/stock/views/stock_quant_views.xml

```diff
@@ -318,7 +318,8 @@
                 <field name="product_id" attrs="{'readonly': [('id', '!=', False)]}" readonly="context.get('single_product', False)" force_save="1" options="{'no_create': True}"/>
                 <field name="product_categ_id" optional="hide"/>
                 <field name="lot_id" groups="stock.group_production_lot"
-                    attrs="{'readonly': ['|', ('id', '!=', False), ('tracking', 'not in', ['serial', 'lot'])]}"
+                    attrs="{'readonly': ['|', ('tracking', 'not in', ['serial', 'lot']), '&amp;', ('id', '!=', False), ('lot_id', '!=', False)]}"
+                    force_save="1"
```

Comment:
Something weird is happening here. For some reason it's still letting me edit the `lot_id` even when id != False (i.e. create a line with no lot/SN + apply (confirm no SN/lot) => edit line's Lot/SN and click off of the line leads to a Usererror)

Reply:
We actually want the empty `lot_id` to be editable when `id` exists, since the `id` is created before clicking on the Apply button. The problem is more that it's still editable after doing the Apply, which I don't know how to solve without introducing a new stored field. Since this part is not exactly a bug but more of an incovenience, maybe this should be changed only in master and not in v15.